src/routes/save-metadata.test.ts

```diff
@@ -30,4 +30,35 @@ describe('POST /metadata', () => {
     expect(response.body).toStrictEqual(JSON.stringify(expectedResponse));
     expect(response.statusCode).toBe(201);
   });
+
+  it('Throws an error if metadata is not strings', async () => {
+    const response = await app.inject({
+      method: 'POST',
+      url: '/metadata',
+      payload: {
+        firstName: 5,
+        lastName: 'Rose',
+      },
+    });
+
+    expect(response.statusCode).toBe(500);
+    expect(JSON.parse(response.body).message).toBe(
+      'Metadata object values have to consist of strings or arrays of strings'
+    );
+  });
+
+  it('Throws an error if metadata is not array of strings', async () => {
+    const response = await app.inject({
+      method: 'POST',
+      url: '/metadata',
+      payload: {
+        lastName: ['Rose', 'Blue', { sneaky: 'object' }],
+      },
+    });
+
+    expect(response.statusCode).toBe(500);
```

Comment:
Feels like this is more of a client rather than a server-side error, maybe a 400 would be more appropriate? https://httpstatuses.com/400